### PR TITLE
EUI-2976: Activity Tracker Socket

### PR DESCRIPTION
**Pull request by paulhowes-HMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-4355

### Change description ###
Activity Tracker with socket.io functionality. Take latest master and merge from EUI-2976, incorporating ccd-case-ui-toolkit socket.io development.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/lib/middleware/proxy.ts [hunk 1]
```diff
@@ -1,83 +1,85 @@
-import {createProxyMiddleware as proxy, Options} from 'http-proxy-middleware';
+import { createProxyMiddleware as proxy, Options } from '@hmcts/http-proxy-middleware';
```
[the remark this answers is not in the document]
Comment (paulhowes-HMCTS) [reply api/lib/middleware/proxy.ts:1]: See https://tools.hmcts.net/jira/browse/EUI-4330 for explanation.

[the remark this answers is not in the document]
Comment (paddybasi) [reply api/lib/middleware/proxy.ts:1]: okay then, lol. Seems like quite the drama

**Review by Mo-Lala84 — DISMISSED** [review]
approved

**Review by paddybasi — DISMISSED** [review]

File: api/proxy.config.ts [hunk 2]
```diff
@@ -1,52 +1,47 @@
...
-  SERVICES_EM_DOCASSEMBLY_API_URL,
-  SERVICES_EM_HRS_API_PATH,
-  SERVICES_ICP_API_URL,
-  SERVICES_MARKUP_API_URL,
-  SERVICES_PAYMENTS_URL,
-  SERVICES_REFUNDS_API_URL,
-  SERVICES_LOCATION_REF_API_URL,
-  SERVICES_ROLE_ASSIGNMENT_API_PATH
+  proxiedReferences
 } from './configuration/references';
```
Comment (GaryScottHMCTS) [inline api/proxy.config.ts:8]: this could be on one line now

**Review by GaryScottHMCTS — APPROVED** [review]

**Pull request closed without merging**